include/ledger_assert.h

```diff
@@ -0,0 +1,159 @@
+#else
+#define MESSAGE_SIZE 0
+#define ASSERT_DISPLAY_MESSAGE(message) \
+    do {                                \
+    } while (0)
+#endif
+
+#if defined(LEDGER_ASSERT_CONFIG_FILE_INFO) && defined(HAVE_LEDGER_ASSERT_DISPLAY)
+#define FILE_OFFSET MESSAGE_OFFSET + MESSAGE_SIZE
+#define FILE_SIZE   50
+void assert_display_file_info(const char *file, unsigned int line);
+#define ASSERT_DISPLAY_FILE_INFO(file, line) assert_display_file_info(file, line)
+#else
+#define FILE_SIZE 0
+#define ASSERT_DISPLAY_FILE_INFO(file, line) \
+    do {                                     \
+    } while (0)
+#endif
+
+#ifdef HAVE_LEDGER_ASSERT_DISPLAY
+#define ASSERT_BUFFER_LEN    LR_AND_PC_SIZE + MESSAGE_SIZE + FILE_SIZE
+#define LEDGER_ASSERT_EXIT() assert_display_exit()
+#else
+#define LEDGER_ASSERT_EXIT() exit_app(true)
+#endif
+
+#if defined(LEDGER_ASSERT_CONFIG_LR_AND_PC_INFO) && defined(HAVE_PRINTF)
+void assert_print_lr_and_pc(int lr, int pc);
+#define ASSERT_PRINT_LR_AND_PC(lr, pc) assert_print_lr_and_pc(lr, pc)
+#else
+#define ASSERT_PRINT_LR_AND_PC(lr, pc) \
+    do {                               \
+    } while (0)
+#endif
+
+#if defined(LEDGER_ASSERT_CONFIG_MESSAGE_INFO) && defined(HAVE_PRINTF)
+void assert_print_message(const char *message);
+#define ASSERT_PRINT_MESSAGE(message) assert_print_message(message)
+#else
+#define ASSERT_PRINT_MESSAGE(message) \
+    do {                              \
+    } while (0)
+#endif
+
+#if defined(LEDGER_ASSERT_CONFIG_FILE_INFO) && defined(HAVE_PRINTF)
+void assert_print_file_info(const char *file, int line);
+#define ASSERT_PRINT_FILE_INFO(file, line) assert_print_file_info(file, line)
+#else
+#define ASSERT_PRINT_FILE_INFO(file, line) \
+    do {                                   \
+    } while (0)
+#endif
+
+#ifdef LEDGER_ASSERT_CONFIG_LR_AND_PC_INFO
+#define LEDGER_ASSERT_LR_AND_PC()                           \
+    do {                                                    \
+        int _lr_address = 0;                                \
+        int _pc_address = 0;                                \
+                                                            \
+        __asm volatile("mov %0, lr" : "=r"(_lr_address));   \
+        __asm volatile("mov %0, pc" : "=r"(_pc_address));   \
+        ASSERT_PRINT_LR_AND_PC(_lr_address, _pc_address);   \
+        ASSERT_DISPLAY_LR_AND_PC(_lr_address, _pc_address); \
+    } while (0)
+#endif
+
+#ifdef LEDGER_ASSERT_CONFIG_MESSAGE_INFO
+#define LEDGER_ASSERT_MESSAGE(message)   \
+    do {                                 \
+        ASSERT_PRINT_MESSAGE(message);   \
+        ASSERT_DISPLAY_MESSAGE(message); \
+    } while (0)
+#endif
+
+#ifdef LEDGER_ASSERT_CONFIG_FILE_INFO
+#define LEDGER_ASSERT_FILE_INFO()                     \
+    do {                                              \
+        ASSERT_PRINT_FILE_INFO(__FILE__, __LINE__);   \
+        ASSERT_DISPLAY_FILE_INFO(__FILE__, __LINE__); \
+    } while (0)
+#endif
```

Comment:
Couldn't this ones always be defined as the ASSERT_PRINT_XXX and ASSERT_DISPLAY_XXX will be void when not needed?
It would also ease the definition of LEDGER_ASSERT.